pkg/util/maven/types.go

```diff
@@ -25,7 +25,7 @@ type ProjectDefinition struct {
 	Project     Project
 	JavaSources map[string]string
 	Resources   map[string]string
-	Env         map[string]string
+	Env         map[string]string // TODO: should we deprecate it ? env are set on deployment
```

Comment:
I'm thinking to a "merge" function, where the configuration at context level is used as base for the configuration at integration level. We can do it when creating the integration, so you're right, this can be removed.

Reply:
It is already done, env, properties, configmap and secretes from the required context are are applied to integration's deployment :)